README.rst

```diff
@@ -1506,79 +1506,366 @@ Get command details
 
     GET /api/v1/controller/device/{device_id}/command/{command_id}/
 
+List device groups
+##################
+
+.. code-block:: text
+
+    GET /api/v1/controller/group/
+
+Create device group
+###################
+
+.. code-block:: text
+
+    POST /api/v1/controller/group/
+
+Get device group detail
+#######################
+
+.. code-block:: text
+
+    GET /api/v1/controller/group/{id}/
+
+Get device group from certificate common name
+#############################################
+
+.. code-block:: text
+
+    GET /api/v1/controller/cert/{common_name}/group/
+
+This endpoint can be used to retrieve group information and metadata by the
+common name of a certificate used in a VPN client tunnel, this endpoint is
+used in layer 2 tunneling solutions for firewall/captive portals.
+
+It is also possible to filter device group by providing organization slug
+of certificate's organization as show in the example below:
+
+.. code-block:: text
+
+    GET /api/v1/controller/cert/{common_name}/group/?org={org1_slug},{org2_slug}
+
+Get device location
+###################
+
+.. code-block:: text
+
+
+    GET /api/v1/controller/device/{id}/location/
+
+
+Create device location
+######################
+
+.. code-block:: text
+
+    POST /api/v1/controller/device/{id}/location/
+
+You can create ``DeviceLocation`` object by using primary
+keys of existing ``Location`` and ``FloorPlan`` objects as shown in
+the example below.
+
+.. code-block:: json
+
+    {
+        "location": "f0cb5762-3711-4791-95b6-c2f6656249fa",
+        "floorplan": "dfeb6724-aab4-4533-aeab-f7feb6648acd",
+        "indoor": "12.342,23.541"
+    }
+
+.. code-block:: text
+
+    curl -X POST \
```

Comment:
let's remember that if we remove the possibility of using POST and allow PUT to create, we will have to update these examples accordingly